src/main/app/src/app/shared/navigation/navigation.service.ts

```diff
@@ -48,27 +48,33 @@ export class NavigationService {
   private handleRouterEvents(e): void {
     if (e instanceof NavigationStart) {
       this.utilService.setLoading(true);
-    } else {
-      if (e instanceof NavigationEnd) {
-        const history = SessionStorageUtil.getItem(
-          NavigationService.NAVIGATION_HISTORY,
-          [],
-        );
-        if (
-          history.length === 0 ||
-          history[history.length - 1] !== e.urlAfterRedirects
-        ) {
-          history.push(e.urlAfterRedirects);
-        }
-        SessionStorageUtil.setItem(
-          NavigationService.NAVIGATION_HISTORY,
-          history,
-        );
-        this.backDisabled.next(history.length <= 1);
+      return;
+    }
+
+    if (e instanceof NavigationError &&
```

Comment:
just wondering if it would be neater to use a `switch` statement here and check the if conditions in the `case` clauses, what's your opinion?